lib/note-content-editor.jsx

```diff
@@ -171,6 +220,29 @@ export default class NoteContentEditor extends React.Component {
 		)
 	}
 
+	handleReturn = () => {
+		// matches lines that start with `- ` or `* `
+		// preceded by 0 or more tab characters
+		const listItemRe = /^\t*[-*]\s/;
+
+		const { editorState } = this.state;
+		const line = getCurrentBlock( editorState ).getText();
+
+		const trimmedLine = line.trim()
+		if ( trimmedLine === '-' || trimmedLine === '*' ) {
+			this.handleEditorStateChange( finishList( editorState ) );
+			return 'handled'
+		}
+
+		const listItemMatch = line.match( listItemRe )
+		if ( listItemMatch ) {
+			this.handleEditorStateChange( continueList( editorState, listItemMatch ) );
+			return 'handled'
+		}
+
+		return 'not-handled';
```

Comment:
are these strings really needed or are they mainly for debugging? I don't think we have to return any values here since this gets asynchronously called on input. we can still return early in 233 and 239 with `return this.handleEditorStateChange( … )` but just leave this function without a proper `return`